src/groupByKey.ts

```diff
@@ -1,7 +1,24 @@
-type GroupsMap<T> = {
+export interface GroupsMap<T> {
   [key: string]: T[];
-};
+}
+
+export type KeyTypes = string | number;
+
+export interface Item {
+  [key: string]: KeyTypes;
+}
+
+export function groupByKey(items: Item[], key: keyof Item): GroupsMap<Item> {
+  const newKeys: KeyTypes[] = [];
+
+  items.forEach((item: Item) => {
+    if (!newKeys.includes(item[key])) {
+      newKeys.push(item[key]);
+    }
+  });
 
-export function groupByKey(items, key) {
-  // write code here;
+  return newKeys.reduce((prev: GroupsMap<Item>, newKey: KeyTypes) => ({
+    ...prev,
+    [newKey]: items.filter((item: Item) => item[key] === newKey),
+  }), {});
```

Comment:
i'm pretty sure you could have it done with just one `.reduce`
try this out and share any concerns.
```
items.reduce((acc, item) => {
    acc[key] = (...acc[key] || []).concat(item)

    return acc;
  }, {})
```

Reply:
Excellent. Thanks 👍